azimuth/config.py

```diff
@@ -131,12 +170,15 @@ class DatasetWarningsOptions(BaseModel):
 class SyntaxOptions(BaseModel):
     short_sentence_max_token: int = 3
     long_sentence_min_token: int = 16
+    spacy_model: str = ""  # Language-based default value set by AzimuthConfig
```

Comment:
Similarly, does that work?
```suggestion
    spacy_model: Union[SupportedSpacyModels, ""] = ""  # Language-based default value set by AzimuthConfig
```

Reply:
That doesn't work...it doesn't like the empty string in the `Union[]`. This works and passes my test. Am I missing any reason why it's not a good idea?

```suggestion
    spacy_model: Union[SupportedSpacyModels, None] = None  # Language-based dynamic default value
```

Reply:
So that is equivalent to `Optional[SupportedSpacyModels]` and makes `None` a valid option. What we disliked about it is that later in the code, the type of `spacy_model` will be `Optional[SupportedSpacyModels]`, even though we know it can only be `SupportedSpacyModels`. For example, the type checking might complain about it being potentially `None` where it can't be, which would be annoying.

If it's not the case in the current situation, and `Optional` works well, awesome, you could indeed make it `Optional[SupportedSpacyModels]`.

Another option is to add something like `default_value = "default_value"` to `SupportedSpacyModels`, have
```suggestion
    spacy_model: SupportedSpacyModels = SupportedSpacyModels.default_value  # Language-based dynamic default value
```
And check for it with an `if else` in the validator.

Reply:
I did the following (took your suggestion but modified it somewhat), and left the validator as is, and it works for my test. Does it seem ok to you?

```
class SupportedSpacyModels(str, Enum):
    null_value = ""
    en_core_web_sm = "en_core_web_sm"
    fr_core_news_md = "fr_core_news_md"

...

class SyntaxOptions(BaseModel):
    short_sentence_max_token: int = 3
    long_sentence_min_token: int = 16
    spacy_model: SupportedSpacyModels = SupportedSpacyModels.null_value  # Language-based default
    subj_tags: List[str] = []  # Language-based dynamic default value
    obj_tags: List[str] = []  # Language-based dynamic default value
```

Reply:
Makes sense to me!

Reply:
I liked the idea of a name like `default_value` as it was telling the user what to expect. I don't think `null_value` conveys that as well.

Reply:
Good idea for the empty string though. I just checked and it works.

If you are curious, that's because `SyntaxOptions` inherits from `str` before `Enum`, so it is `str.__bool__` that is used to determine the truthiness of `values["spacy_model"]` and `values["spacy_model"] = values["spacy_model"] or ...` works. If we were to remove the `str` inheritance, it would stop working.

Reply:
Hmm. I was specifically thinking something other than `default_value` because the default is actually the thing we replace it with later. Is there another name that sounds good to you? Or you still think `default...` is fine because it is the thing that leads to the default being implemented?

Reply:
`use_default` ftw

Reply:
Wait, bad idea, that implies a boolean. Going to do `default_indicator`. Let me know if you have a better idea!

Reply:
Back to `use_default` after discussion.